src/groupeddataframe/utils.jl

```diff
@@ -143,10 +143,28 @@ function refpool_and_array(x::AbstractArray)
         else
             return nothing, nothing
         end
-    elseif x isa AbstractArray{<:Union{Real, Missing}} &&
-        all(v -> ismissing(v) | isinteger(v), x) &&
-        !isempty(skipmissing(x))
-        minval, maxval = extrema(skipmissing(x))
+    elseif x isa AbstractArray{<:Union{Real, Missing}} && length(x) > 0
+        if !(x isa AbstractArray{<:Union{Integer, Missing}})
+            if !all(v -> (ismissing(v) | isinteger(v)) & (!isequal(v, -0.0)), x)
+                return nothing, nothing
+            end
+        end
+        if Missing <: eltype(x)
+            smx = skipmissing(x)
+            y = iterate(smx)
+            y === nothing && return nothing, nothing
+            (v, s) = y
+            minval = maxval = v
+            while true
+                y = iterate(smx, s)
+                y === nothing && break
+                (v, s) = y
+                maxval = max(v, maxval)
+                minval = min(v, minval)
+            end
```

Comment:
Why not just keep using `extrema`? Is this just to avoid checking `isempty(skipmissing(x))` first? If so I'm not sure it's worth it TBH, as it should be relatively cheap compared with the cost of grouping. Also if we wanted to be maximally efficient we should use a single loop in which we would check both whether there's at least one non-missing value and `(ismissing(v) | isinteger(v)) & !isequal(v, -0.0)`.

Reply:
The reason is performance. It is not a huge difference but still noticeable for sizes like 10^8.

> `(ismissing(v) | isinteger(v)) & !isequal(v, -0.0)`

This check is done only for non-`Integer` eltype (and I decided it is OK to run a loop twice, as I am still convinced that it is a rare use-case - people should not `groupby` float values I think).

Reply:
> The reason is performance. It is not a huge difference but still noticeable for sizes like 10^8.

You mean the performance of `isempty(skipmissing(x))`? That cost should be pretty low unless you have 10^8 missing values at the beginning, right?

Reply:
OK - I will revert.